components/tidb_query/src/lib.rs

```diff
@@ -26,6 +26,7 @@
 #![allow(clippy::no_effect)]
 #![feature(box_patterns)]
 #![feature(matches_macro)]
+#![feature(const_generics)]
```

Comment:
Looks like the compiler version is old.

```
   Compiling tidb_query v0.0.1 (/home/guo-shaoge/go/src/github.com/pingcap/tikv/components/tidb_query)
warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> components/tidb_query/src/lib.rs:29:12
   |
29 | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
```

Not sure is it ok to use this feature. @skyzh

Reply:
Yes, we should find another way to implement this.